src/time/span.cr

```diff
@@ -286,6 +286,10 @@ struct Time::Span
   def self.zero
     new(0)
   end
+              
+  def zero?
+    self == typeof(self).zero
```

Comment:
Okay, so there's no point in doing `self == typeof(self).zero` if it's not on a general type. Number is a general type, so it's probably okay to leave it. Time::Span and Complex could probably be more specific, though, since they're concrete types.